plugins/chat/plugin.rb

```diff
@@ -577,14 +614,24 @@ def self.onebox_template
     end
   end
 
+  on(:user_removed_from_group) do |user, group|
+    Jobs.enqueue(:auto_remove_membership_handle_user_removed_from_group, user_id: user.id)
+  end
+
   on(:category_updated) do |category|
     # TODO(roman): remove early return after 2.9 release.
     # There's a bug on core where this event is triggered with an `#update` result (true/false)
     return if !category.is_a?(Category)
-    category_channel = ChatChannel.find_by(auto_join_users: true, chatable: category)
+    category_channel = ChatChannel.find_by(chatable: category)
 
     if category_channel
-      Chat::ChatChannelMembershipManager.new(category_channel).enforce_automatic_channel_memberships
+      if category_channel&.auto_join_users
```

Comment:
Here we’re already sure about having a `category_channel` so we could just do `if category_channel.auto_join_users?`

Reply:
Oh yeah not sure why I kept that, thanks